File: api/src/main/java/com/caibowen/prma/spi/EventAdaptor.java
package com.caibowen.prma.spi;

import com.caibowen.prma.api.model.EventVO;

/**
 * @author BowenCai
 * @since 5-11-2014.
 */
public interface EventAdaptor<V> {

    EventVO from(V otherEvent);
    V to(EventVO vo);
}

File: api/src/main/scala/com/caibowen/prma/api/EventAdaptor.scala
package com.caibowen.prma.api

import com.caibowen.prma.api.model.EventVO

/**
 * @author BowenCai
 * @since  02/12/2014.
 */
trait EventAdaptor[V] {

  def from(otherEvent: V): EventVO

  def to(vo: EventVO): V
}
